osm2streets/src/transform/collapse_intersections.rs

```diff
@@ -130,24 +129,19 @@ fn should_collapse(road1: &Road, road2: &Road) -> Result<()> {
     Ok(())
 }
 
-pub fn collapse_intersection(streets: &mut StreetNetwork, i: NodeID) {
+pub fn collapse_intersection(streets: &mut StreetNetwork, i: IntersectionID) {
```

Comment:
Much of the PR was a fairly mechanical search-and-replace. One of the big changes is here. I thought through this carefully and kind of rewrote the logic, to do less fiddling with IDs. Let me know if it's understandable, or if more comments would be helpful

Reply:
It's straighforward enough enough.